controllers/secretdefinition_controller.go

```diff
@@ -0,0 +1,242 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	secretsmanagerv1alpha1 "github.com/tuenti/secrets-manager/api/v1alpha1"
+	"github.com/tuenti/secrets-manager/backend"
+)
+
+const (
+	// https://golang.org/pkg/time/#pkg-constants
+	timestampFormat = "2006-01-02T15.04.05Z"
+	finalizerName   = "secret.finalizer.secrets-manager.tuenti.io"
+)
+
+// SecretDefinitionReconciler reconciles a SecretDefinition object
+type SecretDefinitionReconciler struct {
+	client.Client
+	Backend              backend.Client
```

Comment:
I'd rename this.... `Backend`.... (but I don't have a proposal really) what is able to this guy?

Reply:
Backend is an iterface for multiple secrets stores. Right now we only support Vault. But changing the name for the sake of just this PR... I'd avoid it.